File: public/core/src/main/java/se/infomaker/frt/moduleinterface/action/ActionHandler.java
package se.infomaker.frt.moduleinterface.action;

import android.content.Context;

import se.infomaker.iap.action.Operation;

public interface ActionHandler {
    /**
     * Performs an operation, if the operation has a return value it is sent back as a String
     */
    String perform(Context context, Operation operation);

    /**
     * Returns true if the interface can perform the operation
     */
    boolean canPerform(Context context, Operation operation);
}

File: public/core/src/main/java/se/infomaker/iap/action/ActionHandler.kt
package se.infomaker.iap.action

import android.content.Context

interface ActionHandler {

    fun isLongRunning(): Boolean = false

    /**
     * Performs an operation, if the operation has a return value it is sent back as a String
     */
    fun perform(context: Context, operation: Operation, onResult: (Result) -> Unit)

    /**
     * Returns true if the interface can perform the operation
     */
    fun canPerform(context: Context, operation: Operation): Boolean
}
